New procedure for Ansible job templates (#2892)
theforeman/foreman-documentation: guides/common/modules/proc_using-ansible-provider-for-package-and-errata-actions.adoc

[id="Using_Ansible_Provider_for_Package_and_Errata_Actions_{context}"]
= Using Ansible provider for package and errata actions

By default, {Project} is configured to use the Script provider templates for remote execution jobs.

> If you prefer using the Ansible provider templates for your remote jobs, you can configure {Project} to use the Ansible provider templates for remote execution features that take a search query input.

Reviewer: Okaaay how about this?

```suggestion
If you prefer using Ansible job templates for your remote jobs, you can configure {Project} to use them by default for remote execution features associated with them.
```

I originally had "features that support them," but I think it's more accurate to say that a template "supports" a feature, rather than the other way around, since you specify the `feature: xxxx` in the template's YAML header.

.Prerequisites
Use remote execution in `ssh` mode.

.Procedure
. In the {ProjectWebUI}, navigate to *Administer* > *Remote Execution Features*.
. Find each feature whose name contains `by_search`.
. Change the job template for these features from `Katello Script Default` to `Katello Ansible Default`.
. Click *Submit*.

> Ansible job templates are now used for remote execution when using the web UI to perform package and errata actions, and when using hammer job-invocation created with remote execution features that take a search query input.

Reviewer: ```suggestion
Ansible job templates are now used for remote execution when using the web UI to perform package and errata actions. They are also used when using "hammer job-invocation create" with the same remote execution features which you altered above.
```

Reviewer: `hammer job-invocation create` is a Hammer command.

[id="Using_Ansible_Provider_for_Package_and_Errata_Actions_{context}"]
= Using Ansible provider for package and errata actions

> By default, {Project} is configured to use the Script provider templates for remote execution jobs.
> If you prefer using Ansible job templates for your remote jobs, you can configure {Project} to use them by default for remote execution features associated with them.
> 
> .Prerequisites
> * Use remote execution in `ssh` mode.

Reviewer 2: ```suggestion
By default, {Project} is configured to use the Script provider templates for remote execution jobs.
If you prefer using Ansible job templates for your remote jobs, you can configure {Project} to use them by default for remote execution features associated with them.
This change can only apply to remote execution in `ssh` mode.
```
I don't think this is an actual prerequisite. But I'm not sure about the wording here.

Reviewer: How is prerequisite defined? If you haven't configured your smart proxy and hosts for `ssh` mode, that means your hosts will not have the SSH keys installed and will never receive the REX job.

Reviewer 2: Well, sure, but you don't need it in order to perform the following procedure, do you?

Reviewer: I suppose you could still set the default templates for those features, yes. But then things would stop working.

Reviewer: How about just a Note, then?

```
Remember that Ansible job templates only work when remote execution is configured for `ssh` mode.`
```

Reviewer 2: That was the idea.

.Procedure
. In the {ProjectWebUI}, navigate to *Administer* > *Remote Execution Features*.
. Find each feature whose name contains `by_search`.
. Change the job template for these features from `Katello Script Default` to `Katello Ansible Default`.
. Click *Submit*.

> Ansible job templates are now used for remote execution when using the web UI to perform package and errata actions.

Reviewer 3: ```suggestion
Ansible job templates are now used for remote execution when using the {ProjectWebUI} to perform package and errata actions.
```

> Ansible job templates are now used for remote execution when using the {ProjectWebUI} to perform package and errata actions.
> They are also used when using `hammer job-invocation create` with the same remote execution features used in the procedure above.

Reviewer 2: ```suggestion
{Project} now uses Ansible provider templates for remote execution jobs by which you can perform package and errata actions.
This applies to job invocations from the {ProjectWebUI} as well as by using Hammer CLI, with the same remote execution features that you have changed.
```
Perhaps something like this?

Reviewer: Why avoid naming `hammer job-invocation create`? I guess it's fine but I feel like the original was easier to read. (I am biased, though.. ;)

Reviewer 2: I think that level of detail is not necessary in this place, but if @bangelic thinks it's better to include it, I'm not against it.

Author: I'll go ahead and keep `hammer job-invocation create` in there.
